backend/common.go

```diff
@@ -91,6 +91,16 @@ func (s *DataSourceInstanceSettings) HTTPClientOptions() (httpclient.Options, er
 		return httpclient.Options{}, err
 	}
 
+	if s.BasicAuthEnabled {
```

Comment:
checked the code, think this is for loki, influx and tempo? they are having basicAuthUser/basicAuthPassword and user/password for the authentication, the code seems good, just don't the reason why we have both of them

Reply:
user/password basic auth fallback is for legacy reasons, influx for example. Reason I kept the logic for jsondata is so app plugins can make use of this